lte/gateway/c/oai/tasks/s1ap/s1ap_mme_handlers.c

```diff
@@ -2565,6 +2565,110 @@ int s1ap_mme_handle_handover_command(
+  // message by changing type and updating the enb_ue_s1ap_id to match that of
+  // the target enb.
+
+  // MME_UE_S1AP_ID
+  S1AP_FIND_PROTOCOLIE_BY_ID(
+      S1ap_ENBStatusTransferIEs_t, ie, container,
+      S1ap_ProtocolIE_ID_id_MME_UE_S1AP_ID, true);
+  if (ie) {
+    mme_ue_s1ap_id = ie->value.choice.MME_UE_S1AP_ID;
+  } else {
+    OAILOG_FUNC_RETURN(LOG_S1AP, RETURNerror);
+  }
+
+  // get the UE and handover state
+  if ((ue_ref_p = s1ap_state_get_ue_mmeid(mme_ue_s1ap_id)) == NULL) {
+    OAILOG_ERROR(
+        LOG_S1AP,
+        "could not get ue context for mme_ue_s1ap_id " MME_UE_S1AP_ID_FMT
+        ", failing!\n",
+        (uint32_t) mme_ue_s1ap_id);
+    OAILOG_FUNC_RETURN(LOG_S1AP, RETURNerror);
+  }
+
+  OAILOG_INFO(
+      LOG_S1AP,
+      "Received eNBStatusTransfer from source enb_id assoc %u for "
+      "ue " MME_UE_S1AP_ID_FMT " to target enb_id %u\n",
+      assoc_id, mme_ue_s1ap_id,
+      ue_ref_p->s1ap_handover_state.target_enb_ue_s1ap_id);
+
+  // set the target eNB_UE_S1AP_ID
+  S1AP_FIND_PROTOCOLIE_BY_ID(
+      S1ap_ENBStatusTransferIEs_t, ie, container,
+      S1ap_ProtocolIE_ID_id_eNB_UE_S1AP_ID, true);
+  if (ie) {
+    ie->value.choice.ENB_UE_S1AP_ID =
+        ue_ref_p->s1ap_handover_state.target_enb_ue_s1ap_id;
+  } else {
+    OAILOG_FUNC_RETURN(LOG_S1AP, RETURNerror);
+  }
+
+  // get the enb_description matching the target_enb_id
+  // retrieve enb_description using hash table and match target_enb_id
+  if ((enb_array = hashtable_ts_get_elements(&state->enbs)) != NULL) {
+    for (idx = 0; idx < enb_array->num_elements; idx++) {
+      target_enb_association =
+          (enb_description_t*) (uintptr_t) enb_array->elements[idx];
+      if (target_enb_association->enb_id ==
+          ue_ref_p->s1ap_handover_state.target_enb_id) {
+        break;
+      }
+    }
+    free_wrapper((void**) &enb_array->elements);
+    free_wrapper((void**) &enb_array);
+    if (target_enb_association->enb_id !=
+        ue_ref_p->s1ap_handover_state.target_enb_id) {
+      OAILOG_ERROR(
+          LOG_S1AP, "No eNB for enb_id %d\n",
+          ue_ref_p->s1ap_handover_state.target_enb_id);
+      OAILOG_FUNC_RETURN(LOG_S1AP, RETURNerror);
+    }
+  }
+
+  // change the message type and enb_ue_s1_id to the target eNB's ID
+  pdu->choice.initiatingMessage.procedureCode =
+      S1ap_ProcedureCode_id_MMEStatusTransfer;
+  pdu->present = S1ap_S1AP_PDU_PR_initiatingMessage;
+
+  // Encode message
+  if (s1ap_mme_encode_pdu(pdu, &buffer, &length) < 0) {
+    OAILOG_ERROR(
+        LOG_S1AP,
+        "Failed to encode MME Configuration Transfer message for enb_id %u\n",
+        ue_ref_p->s1ap_handover_state.target_enb_id);
+    OAILOG_FUNC_RETURN(LOG_S1AP, RETURNerror);
+  }
+
+  bstring b = blk2bstr(buffer, length);
+  free(buffer);
+
+  s1ap_mme_itti_send_sctp_request(
+      &b, target_enb_association->sctp_assoc_id, 0x0,
+      ue_ref_p->mme_ue_s1ap_id);  // replace 0x0 with target ue stream
```

Comment:
looks like you did forget to do what the comment on this line is stating?

Reply:
YIkes! Good catch...